lib/builder.js

```diff
@@ -104,13 +111,15 @@ Builder.prototype.build = function() {
           })
       })
   })
-  return promise
+  return promise;
 }
 
 // Destructor-like method. Cleanup is synchronous at the moment, but in the
 // future we might change it to return a promise.
 Builder.prototype.cleanup = function() {
+  this._canceled = true
   this.builderTmpDirCleanup()
```

Comment:
I don't think it's correct to try removing tmp dirs with `builderTmpDirCleanup` while the build is still in progress. We'd need to wait for the current step to finish first.